jenkins_pipelines/environments/common/pipeline-reference.groovy

```diff
@@ -0,0 +1,44 @@
+def run(params) {
+    timestamps {
+        deployed = false
+        env.resultdir = "${WORKSPACE}/results"
+        env.resultdirbuild = "${resultdir}/${BUILD_NUMBER}"
+        env.common_params = "--outputdir ${resultdir} --tf ${params.tf_file} --gitfolder ${resultdir}/sumaform"
+        try {
+            stage('Clone sumaform and susemanager-ci') {
+                // Create a directory for all the outputs belonging to this build number
+                sh "mkdir -p ${resultdir}"
+                git url: params.terracumber_gitrepo, branch: params.terracumber_ref
+                dir("susemanager-ci") {
+                    checkout scm
+                }
+                // Create a directory for  to place the directory with the build results (if it does not exist)
+                sh "mkdir -p ${resultdir}"
+                // Clone sumaform
+                sh "set +x; source /home/jenkins/.credentials set -x; ./terracumber-cli ${common_params} --gitrepo ${params.sumaform_gitrepo} --gitref ${params.sumaform_ref} --runstep gitsync"
+            }
+            stage('Deploy') {
+                // Provision the environment
+                if (params.terraform_bin_init) {
+                    env.TERRAFORM_INIT = '--init'
+                } else {
+                    env.TERRAFORM_INIT = ''
+                }
+                sh "set +x; source /home/jenkins/.credentials set -x; TERRAFORM=${params.terraform_bin} TERRAFORM_PLUGINS=${params.terraform_bin_plugins} ./terracumber-cli ${common_params} --logfile ${resultdirbuild}/sumaform.log ${env.TERRAFORM_INIT} --taint '.*(domain|main_disk).*' --runstep provision"
+                deployed = true
+            }
+        }
+        finally {
+            stage('Get results') {
+                if (!deployed) {
```

Comment:
Going to the details, not asking to do, we will not support something that Jenkins uses to do, which is sent an email when the job came back to green, after a failed job.

Reply:
That's right. We never supported it. For that we should use the Jenkins module. Still we should need this step in case we want to run from m226 if Jenkins is dead (as it happened in the past).